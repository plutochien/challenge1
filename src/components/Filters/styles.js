import styled from 'styled-components';

export const Button = styled.button`
  color: white;
  background-color: ${props => (props.isSelected ? '#17a2b8' : '#6c757d')};
  border-color: ${props => (props.isSelected ? '#17a2b8' : '#6c757d')};
  text-align: center;
  vertical-align: middle;
  font-weight: 400;
  border: 1px solid transparent;
  padding: 6px 12px;
  font-size: 16px;
  line-height: 1.5;
  border-radius: 4px;
  margin: 5px;
  cursor: pointer;
`;
