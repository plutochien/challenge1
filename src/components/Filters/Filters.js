import PropTypes from 'prop-types';
import React, { useState } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import { setFilter } from '../../actions/todoAction';
import { FILTERS, FILTERS_DESC } from '../../util/constatns';
import { Button } from './styles';

const Filters = ({ setFilter }) => {
  const [currentFilter, setCurrentFilter] = useState(FILTERS.SHOW_ALL);
  const handleSetFilter = filter => {
    setFilter(filter);
    setCurrentFilter(filter);
  };
  return (
    <div>
      <Button
        isSelected={currentFilter === FILTERS.SHOW_ALL}
        onClick={() => handleSetFilter(FILTERS.SHOW_ALL)}
      >
        {FILTERS_DESC.SHOW_ALL}
      </Button>
      <Button
        isSelected={currentFilter === FILTERS.SHOW_COMPLETE}
        onClick={() => {
          handleSetFilter(FILTERS.SHOW_COMPLETE);
        }}
      >
        {FILTERS_DESC.SHOW_COMPLETE}
      </Button>
      <Button
        isSelected={currentFilter === FILTERS.SHOW_INCOMPLETE}
        onClick={() => {
          handleSetFilter(FILTERS.SHOW_INCOMPLETE);
        }}
      >
        {FILTERS_DESC.SHOW_INCOMPLETE}
      </Button>
    </div>
  );
};

Filters.propTypes = {
  setFilter: PropTypes.func.isRequired,
};

const mapStateToProps = (state, ownProps) => ({
  active: ownProps.filter === state.filter,
});

const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      setFilter,
    },
    dispatch,
  );

export default connect(mapStateToProps, mapDispatchToProps)(Filters);
