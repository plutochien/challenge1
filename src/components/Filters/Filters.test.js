import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import Filters from './Filters';
const mockStore = configureStore([]);

describe('Filters Component', () => {
    let store;
    let component;
    beforeEach(() => {
        store = mockStore({
            todos: {
                list: [
                    {
                        id: 7,
                        taskText: 'task1',
                        isComplete: true,
                        priority: '0'
                    },
                    {
                        id: 6,
                        taskText: 'task2',
                        isComplete: false,
                        priority: '1'
                    },
                    {
                        id: 1,
                        taskText: 'task3',
                        isComplete: true,
                        priority: '2'
                    },
                ],
            },
            filter: 'SHOW_ALL',
        });
        store.dispatch = jest.fn();
        component = renderer.create(
            <Provider store={store}>
                <Filters setFilter={jest.fn}/>
            </Provider>,
        );

    });

    it('should render with given state from Redux store', () => {
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('should dispatch an action on click All button', () => {
        renderer.act(() => {
        component.root.findAllByType('button')[0].props.onClick();
      });
      expect(store.dispatch).toHaveBeenCalledTimes(1);
    });

    it('should dispatch an action on click Complete button', () => {
        renderer.act(() => {
            component.root.findAllByType('button')[1].props.onClick();
        });
        expect(store.dispatch).toHaveBeenCalledTimes(1);
    });

    it('should dispatch an action on click Incomplete button', () => {
        renderer.act(() => {
            component.root.findAllByType('button')[2].props.onClick();
        });
        expect(store.dispatch).toHaveBeenCalledTimes(1);
    });
});
