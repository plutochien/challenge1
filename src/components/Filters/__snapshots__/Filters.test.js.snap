// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters Component should render with given state from Redux store 1`] = `
<div>
  <button
    className="sc-bdfBwQ iaWOAt"
    onClick={[Function]}
  >
    All
  </button>
  <button
    className="sc-bdfBwQ gGBLyl"
    onClick={[Function]}
  >
    Complete
  </button>
  <button
    className="sc-bdfBwQ gGBLyl"
    onClick={[Function]}
  >
    Incomplete
  </button>
</div>
`;
